### ISSUE TITLE
Tab Completion broken for directories with spaces

### ISSUE THREAD

**Issue by jaraco** [xonsh/xonsh#2769]
## xonfig

~ $ xonfig

```
+------------------+----------------------+
| xonsh            | 0.7.3                |
| Git SHA          | 55d69ee6             |
| Commit Date      | Aug 11 10:22:24 2018 |
| Python           | 3.7.0                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | None                 |
| shell type       | readline             |
| pygments         | None                 |
| on posix         | True                 |
| on linux         | False                |
| on darwin        | True                 |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

## Expected Behavior
Tab completion should work even for files with spaces.

## Current Behavior
When using xonsh, frequently tab completion creates broken syntax, especially when spaces are involved in cwd or item being expanded. Sometime between 0.6.7 and 0.7.3, the situation improved, but issues still remain. For example:

```
~ $ cd draft
draft $ mkdir 'Virtualbox VMs'
draft $ mkdir 'VMWare VMs'
draft $ os.listdir()
['Virtualbox VMs', 'VMWare VMs']
```

Then, typing `cd VM^t` yields:

```
draft $ cd VMs/'
```

Which obviously doesn't work.

**Comment by jaraco** [comment]
The issue goes away with `prompt_toolkit` installed and enabled.

**Comment by jaraco** [comment]
Is using prompt_toolkit the preferred usage? If so, are there instructions on enabling it by default?

**Comment by scopatz** [comment]
Hi @jaraco - yes, prompt-toolkit is the preffered frontend.  It is enabled by default if it is installed (since this is what `best` resolves to). There are docs for `$SHELL_TYPE` [here](https://xon.sh/envvars.html#shell-type) or you can see the help in `xonsh --help`

**Comment by scopatz** [comment]
It is weird that this breaks on readline, though.

**Comment by jaraco** [comment]
> It is enabled by default if it is installed.

I suspected this would be the case, so I tried it, but it didn't take effect the first time I tried (although I may have launched xonsh from a readline xonsh, which may have affected it, or I may have not launched a new xonsh correctly at all). To confirm, though, I am seeing prompt-toolkit used by default now.

**Comment by scopatz** [comment]
hey @jaraco - finally figured this out.  It was a truth table error. The fix is in #2813
